Microsoft.AVS.Management/Microsoft.AVS.Management.psm1

```diff
@@ -255,6 +255,152 @@ function Get-CertificateFromDomainController {
+            foreach($url in $urls) {
+                Write-Host "* Checking LDAP URL: $url"
+                
+                # Check URL looks okay:
+                # i.e. ldaps://ldap1.ldap.avs.azure.com
+                if($url.ToLower() -match '(?<protocol>ldap|ldaps)://(?<hostname>[a-z0-9\.]+)(?<portspec>$|:[0-9]+)') {
+                    $ldap_protocol = $Matches.protocol
+                    $ldap_hostname = $Matches.hostname
+                    $ldap_portspec = $Matches.portspec
+                    Write-Host "  LDAP Protocol:        $ldap_protocol"
+                    Write-Host "  LDAP Server Hostname: $ldap_hostname"
+                    Write-Host "  LDAP Port Specified:  $ldap_portspec"
+
+                    # Check if the LDAP hostname is in /etc/hosts
+                    try {
+                        $Command = "grep $ldap_hostname /etc/hosts"
+                        $SSHRes = Invoke-SSHCommand -Command $Command -SSHSession $SSH_Sessions['VC'].Value
+                    }
+                    catch {
+                        throw "ERROR: Unable to execute grep command on vCenter."
+                    }
+                    $SSHOutput = $SSHRes.Output | out-string        
+                    Write-Host "* vCenter /etc/hosts hostname resolution check returned: $($SSHRes.Output)"
+
+                    # Call Host to check DNS resolution of LDAP server from vCenter"
+                    try {
+                        $Command = "host $ldap_hostname"
+                        $SSHRes = Invoke-SSHCommand -Command $Command -SSHSession $SSH_Sessions['VC'].Value
+                    }
+                    catch {throw "ERROR: Unable to execute host command on vCenter."}
+                    Write-Host "* vCenter DNS hostname resolution check returned: $($SSHRes.Output)"
+
+                    # Now let's look at the port numbers            
+                    if($ldap_portspec -ne "") {
+                        $ldap_port = $ldap_portspec -match ":([0-9]+)"
+                        Write-Host "  LDAP Port number:      $ldap_port"
+                    } else {
+                        switch($ldap_protocol) {
+                            "ldap"  {$ldap_port = 389}
+                            "ldaps" {$ldap_port = 636}
+                            "default" {$ldap_port = -1}
+                        }
+                        Write-Host "* LDAP Port to test:    $ldap_port"
+                    }
+
+                    # Call NetCat to test if the LDAP port is open 
+                    try {
+                        $Command = "nc -vz $ldap_hostname $ldap_port"
+                        $SSHRes = Invoke-SSHCommand -Command $Command -SSHSession $SSH_Sessions['VC'].Value
+                    }
+                    catch {throw "ERROR: Unable to execute nc command on vCenter."}
+                    
+                    if($SSHRes.ExitStatus -eq 1) {
+                        Write-Host "* vCenter-to-LDAP TCP test FAILED!"
```

Comment:
Let's cut the drama - no exclamations, and use correct output stream, i.e. `Write-Error` if we encounter error. 
If the error is terminal, i.e. TCP test failed, then use `throw` instead to terminate early.